test/jdk/javax/security/auth/Subject/CallAsWithScopedValue.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8296244
+ * @enablePreview
+ * @summary Implement Subject.current and Subject.callAs using scoped values.
+ *      Need @enablePreview to use StructuredTaskScope.
+ * @run main/othervm -Djava.security.manager=allow CallAsWithScopedValue false
+ * @run main/othervm -Djava.security.manager=disallow CallAsWithScopedValue true
+ */
+import com.sun.security.auth.UserPrincipal;
+
+import javax.security.auth.Subject;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.StructuredTaskScope;
+
+public class CallAsWithScopedValue {
+
+    private static Map results = new ConcurrentHashMap<Integer,Boolean>();
+
+    public static void main(String[] args) throws Exception {
+
+        boolean usv = Boolean.parseBoolean(args[0]);
+
+        Subject subject = new Subject();
+        subject.getPrincipals().add(new UserPrincipal("Duke"));
+
+        // Always observable in the same thread
+        Subject.callAs(subject, () -> check(0, Subject.current(), "Duke"));
+
+        // Observable in the same thread in ACC mode, but not in the SV mode
```

Comment:
Should this comment actually say "Observable in a new platform thread in ACC mode, but not in the SV mode".